packages/playground/src/app/default-query-function-page/default-query-function-page.component.ts

```diff
@@ -0,0 +1,69 @@
+import { CommonModule } from '@angular/common';
+import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
+import { QueryProvider } from '@ngneat/query';
+import { SubscribeModule } from '@ngneat/subscribe';
+import { switchMap, filter, BehaviorSubject } from 'rxjs';
+
+interface Post {
+  userId: number;
+  id: number;
+  title: string;
+  body: string;
+}
+
+@Component({
+  selector: 'ng-query-default-query-function-page',
+  standalone: true,
+  imports: [CommonModule, SubscribeModule],
+  template: `
+    <p>
+      In this example we are creating queries by only supplying a query key
+      while relying on defaultQueryFn defined in our QueryClientConfig.
+    </p>
+    <p>
+      As you visit the posts below, you will notice them in a loading state the
+      first time you load them. However, after you return to this list and click
+      on any posts you have already visited again, you will see them load
+      instantly and background refresh right before your eyes!
+      <strong>
+        (You may need to throttle your network speed to simulate longer loading
+        sequences)
+      </strong>
+    </p>
+
+    <ng-container *ngIf="selectedPostId$ | async; else postListTpl">
+      <ng-container *subscribe="selectedPost$ as selectedPost">
+        <button class="btn btn-link" (click)="selectedPostId$.next(null)">
+          Back
+        </button>
+        <div class="p-4">
+          <h2>{{ selectedPost.data?.title }}</h2>
+          <p>{{ selectedPost.data?.body }}</p>
+        </div>
+      </ng-container>
+    </ng-container>
+
+    <ng-template #postListTpl>
+      <ng-container *subscribe="posts$ as posts">
+        <button
+          *ngFor="let post of posts.data"
+          class="btn btn-link block"
+          (click)="selectedPostId$.next(post.id)"
+        >
+          {{ post.title }}
+        </button>
+      </ng-container>
+    </ng-template>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DefaultQueryFunctionPageComponent {
+  useQuery = inject(QueryProvider);
+  selectedPostId$ = new BehaviorSubject<number | null>(null);
+
+  posts$ = this.useQuery<Post[]>(['/posts']).result$;
+  selectedPost$ = this.selectedPostId$.asObservable().pipe(
+    filter(Boolean),
```

Comment:
You can use a Subject and remove the filter

Reply:
I can use the Subject but without filter it triggers useQuery with null.

Reply:
Why null?

Reply:
Tried to replicate the example from Tanstack, when selectedId is null we show a list of posts.

Reply:
Ok cool

Reply:
had to use ReplaySubject